infra/modules/gcp-psoxy-rest/main.tf

```diff
@@ -46,44 +46,54 @@ resource "google_service_account_iam_member" "act_as" {
 }
 
 
-resource "google_cloudfunctions_function" "function" {
+resource "google_cloudfunctions2_function" "function" {
   name        = "${var.environment_id_prefix}${var.instance_id}"
   description = "Psoxy Connector - ${var.source_kind}"
-  runtime     = "java17"
-  project     = var.project_id
-  region      = var.region
-
-  trigger_http                 = true
-  https_trigger_security_level = "SECURE_ALWAYS"
-  available_memory_mb          = var.available_memory_mb
-  source_archive_bucket        = var.artifacts_bucket_name
-  source_archive_object        = var.deployment_bundle_object_name
-  entry_point                  = "co.worklytics.psoxy.Route"
-  service_account_email        = var.service_account_email
-  labels                       = var.default_labels
-  docker_registry              = "ARTIFACT_REGISTRY"
-  docker_repository            = var.artifact_repository_id
-
-  environment_variables = merge(
-    local.required_env_vars,
-    var.path_to_config == null ? {} : yamldecode(file(var.path_to_config)),
-    var.environment_variables,
-    var.config_parameter_prefix == null ? {} : { PATH_TO_SHARED_CONFIG = var.config_parameter_prefix },
-    var.config_parameter_prefix == null ? {} : { PATH_TO_INSTANCE_CONFIG = "${var.config_parameter_prefix}${replace(upper(var.instance_id), "-", "_")}_" },
-  )
-
-  dynamic "secret_environment_variables" {
-    for_each = var.secret_bindings
-    iterator = secret_environment_variable
-
-    content {
-      key        = secret_environment_variable.key
-      project_id = data.google_project.project.number
-      secret     = secret_environment_variable.value.secret_id
-      version    = secret_environment_variable.value.version_number
+
+  project  = var.project_id
+  location = var.region
+
+  build_config {
+    runtime           = "java21"
+    docker_repository = var.artifact_repository_id
+    entry_point       = "co.worklytics.psoxy.Route"
+
+    source {
+      storage_source {
```

Comment:
I was thinking if this is still required, but maybe we can start using artifact repository if possible instead of using buckets for this. We have a todo for that: https://github.com/Worklytics/psoxy/pull/879/files#diff-1863f04a027d498e88a5f196aab09d85de06ea0cfce398427b4444cc07abcd1aL34 so maybe we can do this in this or next version?

Reply:
yeah, i saw that and looked at it a bit.

it would appear that cloud functions is creating Artifact Registry underneath automatically; perhaps if we create and then specify one explicitly, that's avoided?

Reply:
I think that was the original idea, but we faced with the update of google provider to support it to have it with the right clean policies.

Ideally, I'd like to try with artifact repository to see if we can improve deployment time with multiple functions too.